src/Orleans.TestingHost/Utils/TestingUtils.cs

```diff
@@ -74,19 +74,21 @@ public static ILoggerFactory CreateDefaultLoggerFactory(string filePath, LoggerF
         /// <summary> Run the predicate until it succeed or times out </summary>
         /// <param name="predicate">The predicate to run</param>
         /// <param name="timeout">The timeout value</param>
+        /// <param name="delayOnFail">The time to delay next call upon failure</param>
         /// <returns>True if the predicate succeed, false otherwise</returns>
-        public static async Task WaitUntilAsync(Func<bool,Task<bool>> predicate, TimeSpan timeout)
+        public static async Task WaitUntilAsync(Func<bool,Task<bool>> predicate, TimeSpan timeout, TimeSpan? delayOnFail = null)
         {
+            delayOnFail = delayOnFail ?? TimeSpan.FromSeconds(1);
             var keepGoing = new[] { true };
             Func<Task> loop =
                 async () =>
                 {
                     bool passed;
                     do
                     {
-                        // need to wait a bit to before re-checking the condition.
-                        await Task.Delay(TimeSpan.FromSeconds(1));
                         passed = await predicate(false);
+                        // need to wait a bit to before re-checking the condition.
+                        if (!passed) await Task.Delay(delayOnFail.Value);
```

Comment:
I feel that we add `keepGoing[0]` to the guard here.

Reply:
Definitely debatable.  I'd considered including it.

I didn't because the behavior in that case would be for the final predicate(true) call to be made immediately after the last failed call without any delay, which was the original behavior, but probably not what we want.

What is the behavior we want?  I think we want the call to be made repeatedly until it succeeds, with a delay before each retry.  If the timeout fires, I'd think we'd still want to have a delay before the retry.

Do you disagree?